src/platform/K32W/NFCManagerImpl.cpp

```diff
@@ -0,0 +1,292 @@
+            if (!NTAG_ReadBytes(sInstance.ntagDriverHandleInstance, 0, &byte0, sizeof(byte0)))
+            {
+                i2cAddrFound = TRUE;
+            }
+            else
+            {
+                /* Loop to try to find a valid i2c address */
+                for (i = 0; i < 0xFF; i++)
+                {
+                    if (i == 2 || i == 0x55) /* Skip default and 0x02 I2C address */
+                    {
+                        continue;
+                    }
+
+                    NTAG_SetNtagI2cAddress(sInstance.ntagDriverHandleInstance, i);
+                    if (!NTAG_ReadBytes(sInstance.ntagDriverHandleInstance, 0, &byte0, sizeof(byte0)))
+                    {
+                        i2cAddrFound = TRUE;
+                        break;
+                    }
+                }
+            }
+
+            if (!i2cAddrFound)
+            {
+                ntagErr = E_APP_NTAG_READ_ERROR;
+                break;
+            }
+        }
+
+        ntagErr = AppNtagEepromUnlockThenWrite(0);
+
+    } while (0);
+
+    return ntagErr;
+}
+
+bool NFCManagerImpl::AppNtagEepromWrite(uint8_t originalSize)
+{
+    bool wasWritten      = FALSE;
+    uint32_t ndefSize    = AppNdefUriRecordGetSize(sInstance.ndefUriRecord);
+    uint32_t addrToWrite = NTAG_I2C_BLOCK_SIZE;
+    uint8_t buf[4];
+    uint8_t terminatorTLV = 0xFE;
+
+    do
+    {
+        if (!sInstance.ndefUriRecord.payloadLen)
+        {
+            if (NTAG_WriteBytes(sInstance.ntagDriverHandleInstance, addrToWrite, (uint8_t *) &(sInstance.ndefUriRecord),
+                                originalSize + sizeof(terminatorTLV)))
+            {
+                break;
+            }
+        }
+        else
+        {
+            buf[0] = 0x3;
+            buf[1] = ndefSize;
+            if (NTAG_WriteBytes(sInstance.ntagDriverHandleInstance, addrToWrite, buf, 2))
+            {
+                break;
+            }
+
+            addrToWrite += 2;
+
+            if (NTAG_WriteBytes(sInstance.ntagDriverHandleInstance, addrToWrite, (uint8_t *) &(sInstance.ndefUriRecord), ndefSize))
+            {
+                break;
+            }
+
+            addrToWrite += ndefSize;
+
+            if (NTAG_WriteBytes(sInstance.ntagDriverHandleInstance, addrToWrite, &terminatorTLV, 1))
+            {
+                break;
+            }
+        }
+
+        wasWritten = TRUE;
+    } while (0);
+
+    return wasWritten;
+}
+
+NFCManagerImpl::eAppNtagError NFCManagerImpl::AppNtagEepromUnlockThenWrite(uint8_t originalSize)
+{
+    eAppNtagError ntagErr = E_APP_NTAG_NO_ERROR;
+
+    do
+    {
+        /* Unlock write access */
+        ntagErr = AppNtagUnlockWriteAccess();
+        if (ntagErr != E_APP_NTAG_NO_ERROR)
+        {
+            break;
+        }
+
+        /* Write the smart poster value */
+        if (!AppNtagEepromWrite(originalSize))
+        {
+            ntagErr = E_APP_NTAG_WRITE_ERROR;
+            break;
```

Comment:
nit: Should you break or LockWriteAccess? (maybe it's okay..).

Reply:
Thanks for spotting this. Fixed it!